internal/testcontainersdocker/docker_host.go

```diff
@@ -29,33 +45,101 @@ func DefaultGatewayIP() (string, error) {
 	return ip, nil
 }
 
-// Extracts the docker host from the context, or returns the default value
-func ExtractDockerHost(ctx context.Context) (dockerHostPath string) {
-	if dockerHostPath = os.Getenv("TESTCONTAINERS_DOCKER_SOCKET_OVERRIDE"); dockerHostPath != "" {
-		return dockerHostPath
+// ExtractDockerHost Extracts the docker host from the different alternatives, caching the result to avoid unnecessary
+// calculations.
+// The possible alternatives are:
+//   1. DOCKER_HOST environment variable
+//   2. TESTCONTAINERS_DOCKER_SOCKET_OVERRIDE environment variable
+//   3. Docker socket path from context
+//   4. Docker socket path from the default docker socket path
+//   5. Docker socket path from the "docker.host" property in the ~/.testcontainers.properties file
+//   6. Rootless docker socket path
+//   If none of the above alternatives are found, an empty string is returned
+func ExtractDockerHost(ctx context.Context) string {
+	dockerHostOnce.Do(func() {
+		dockerHostCache = extractDockerHost(ctx)
+	})
+
+	return dockerHostCache
+}
+
+// extractDockerHost Extracts the docker host from the different alternatives, without caching the result.
+// This internal method is handy for testing purposes.
+func extractDockerHost(ctx context.Context) string {
+	socketPathFns := []func(context.Context) (string, error){
+		dockerHostFromEnv,
+		dockerSocketOverridePath,
+		dockerSocketFromContext,
+		dockerSocketPath,
+		dockerSocketFromProperties,
+		rootlessDockerSocketPath,
```

Comment:
I am uncertain if I have missed something, but it appears to me that the Docker socket override configuration is being used as a Docker host. However, this is wrong. The Docker socket override represents a distinct configuration (`DOCKER_HOST` is not equivalent to `TESTCONTAINERS_DOCKER_SOCKET_OVERRIDE`). It is important to note that these two configurations do not serve as replacements for one another.

The Docker host, serves as the endpoint for Testcontainers communication. While the socket path is utilized by containers such as the Resource Reaper (Ryuk) as mount point.

It is worth noting that due to the possibility of containers running within a virtual machine (VM aka Docker Desktop), the host and socket configurations may differ:

```
DOCKER_HOST=unix:///$HOME/.docker/run/docker.sock
TESTCONTAINERS_DOCKER_SOCKET_OVERRIDE=/var/run/docker.sock
```